### PR TITLE
FW-1916 Click event to send chat data to google analytics

### PR DESCRIPTION
**Pull request by Ojashwa**
### What do you want to achieve?
- Send chat widget events to Google Analytics and access to some custom event

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- Log in to the [website](https://synthesizer-harmonica-8st7.squarespace.com/) using password:-123, GA script is already loaded in this demo webpage.

- You can use your own GA tracking ID (UA-xxxx-x) pass it to your app settings and try to send events (click on widget open /close, rich button click, FAQ click ) list of all [events](https://kommunicate.atlassian.net/wiki/spaces/PD/pages/591626293/Google+Analytics+Event+for+Chat+Widget).

- Google Analytics might take 8-10 hr to update the events for the respective GA tracking ID, you can track a particular event by the help of  `Google Analytics Debugger` chrome extension 
---
image

---
### For user custom events you can subscribe to them in **_onInIt function_** as shown below.

> (function (d, m) {
      var kommunicateSettings = {
        appId: "8ced8a81108c654234a393115cc2ea4f",
        locShare: true,
        onInit: function () {
          var events = {
            onChatWidgetOpen: function (resp) {
              console.log(resp);
            },
            onChatWidgetClose: function (resp) {
              console.log(resp);
            },
            onFaqClick: function (resp) {
              console.log(resp);
            },
            onStartNewConversation: function (resp) {
              console.log(resp);
            },
            onAttachmentClick: function (resp) {
              console.log(resp);
            },
            onVoiceIconClick: function (resp) {
              console.log(resp);
            },
            onLocationIconClick: function (resp) {
              console.log(resp);
            },
            onRateConversationClick: function (resp) {
              console.log(resp);
            },
            onRateConversationEmoticonsClick: function (resp) {
              console.log(resp);
            },
            onSubmitRatingClick: function (resp) {
              console.log(resp);
            },
            onRestartConversationClick: function (resp) {
              console.log(resp);
            },
            onGreetingMessageNotificationClick: function (resp) {
              console.log(resp);
            },
            onNotificationClick: function (resp) {
              console.log(resp);
            },
            onRichMessageButtonClick: function (resp) {
              console.log(resp);
            },
            onShowResolvedClick: function (resp) {
              console.log(resp);
            },
          };
          Kommunicate.subscribeToEvents(events);
        },
      };
      var s = document.createElement("script");
      s.type = "text/javascript";
      s.async = true;
      s.src = "http://localhost:3030/v2/kommunicate.app";
      var h = document.getElementsByTagName("head")[0];
      h.appendChild(s);
      window.kommunicate = m;
      m._globals = kommunicateSettings;
    })(document, window.kommunicate || {});

- For example, if you subscribe to an event called ` onFAQclick` in the script then you get an object of event property.
---
image
---
image

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/km-utils.js [hunk 1]
```diff
@@ -201,6 +201,23 @@ Kommunicate = {
     internetStatus: true,
 };
 
+KM_WidgetEvents={
```
Comment (devashishmamgain) [inline webplugin/js/app/km-utils.js:204]: can you move all analytics related code to a separate file, and keep the minimum code in other files just for calling the respective function.
this will make the code easy to understand

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-utils.js (hunk 1 above)
Comment (Ojashwa) [reply webplugin/js/app/km-utils.js:204]: for the existing event (i.e. `onMessageSent`) keeping them the same as they are, and for new will create a new file.

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 2]
```diff
@@ -749,6 +749,11 @@ Kommunicate.richMsgEventHandler = {
         console.log('passenger detail submitted');
     },
     processQuickReplies: function (e) {
+        window.KM_WidgetEvents.onRichMessageButtonClick('Rich button clicked');
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:752]: check if the google analytics function can be made from KM_WidgetEvents itself, so that all places we are making only 1 common function call

File: webplugin/js/app/kommunicate-ui.js [hunk 3]
```diff
@@ -778,6 +790,18 @@ KommunicateUI = {
         }
     },
     triggerCSAT: function () {
+        window.KM_WidgetEvents.onRateConversationClick(
+            JSON.stringify({
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:794]: is stringify required?

File: webplugin/js/app/km-widget-custom-events.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ // Kommunicate custome events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+KM_WidgetEvents = {
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:6]: use proper naming convention

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-ui.js (hunk 3 above)
Comment (Ojashwa) [reply webplugin/js/app/kommunicate-ui.js:794]: are we returning a JSON string or an object to users when they subscribe to a specific event?

Comment (devashishmamgain) [reply webplugin/js/app/kommunicate-ui.js:794]: return object

Comment (Ojashwa) [reply webplugin/js/app/kommunicate-ui.js:794]: okay

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 5]
```diff
@@ -749,6 +749,16 @@ Kommunicate.richMsgEventHandler = {
         console.log('passenger detail submitted');
     },
     processQuickReplies: function (e) {
+        window.kmWidgetEvents.onRichMessageButtonClick({
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:752]: avoid calling 2 things from the main code. call a single thing related to events subsciption from here, and then from the events thing, call both events callback and the google analytics code

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-ui.js [hunk 6]
```diff
@@ -351,6 +351,11 @@ KommunicateUI = {
 
         // On Click of FAQ button the FAQ List will open.
         $applozic(d).on('click', '#km-faq', function () {
+            window.kmWidgetEvents.eventTracking({
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:354]: use window.kmWidgetEvents.onFaqClick();

File: webplugin/js/app/km-widget-custom-events.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
+/**
+ // Kommunicate custome events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+kmWidgetEvents = {
+    onChatWidgetOpen: function (resp) {},
+    onChatWidgetClose: function (resp) {},
+    onFaqClick: function (resp) {},
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:9]: call the GA code from here

onFaqClick: function() { 
    sendEventToGoogleAnalytics(eventMapping['FAQ Menu']);
}

File: webplugin/js/app/km-widget-custom-events.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
+/**
+ // Kommunicate custome events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:5]: create event mapping here
  var eventMapping = { 'FAQ Menu': {eventCateogry: 'Kommunicate',
                eventAction: 'Click',
                eventLabel: 'FAQ Menu'} ,
                // ... add same for all events...
                
                };

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
added comments for 'FAQ Click' event, check that. then do the same for all events

File: webplugin/js/app/km-widget-custom-events.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+kmWidgetEvents = {
+    onChatWidgetOpen: function (resp) {},
+    onChatWidgetClose: function (resp) {},
+    onFaqClick: function (resp) {},
+    onStartNewConversation: function (resp) {},
+    onAttachmentClick: function (resp) {},
+    onVoiceIconClick: function (resp) {},
+    onVoiceIconClick: function (resp) {},
+    onLocationIconClick: function (resp) {},
```
Comment (Prochnost55) [inline webplugin/js/app/km-widget-custom-events.js:14]: we can remove duplicate

File: webplugin/js/app/kommunicate-ui.js [hunk 10]
```diff
@@ -1257,6 +1278,17 @@ KommunicateUI = {
             bannerHeading &&
                 (bannerHeading.innerHTML =
                     MCK_LABELS['filter.conversation.list'].ALL_CONVERSATIONS);
+            if (
+                bannerAction &&
+                bannerAction.innerHTML ==
+                    MCK_LABELS['filter.conversation.list'].SHOW_RESOLVED
+            ) {
```
Comment (Prochnost55) [inline webplugin/js/app/kommunicate-ui.js:1285]: can we use innerText ?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 11]
```diff
@@ -6638,7 +6759,7 @@ var userOverride = {
                                         data.groupFeeds[0] &&
                                         (updateConversationHeaderParams.imageUrl =
                                             data.groupFeeds[0].imageUrl);
-                                    !params.isConversationInWaitingQueue && mckMessageService.processOnlineStatusChange(
+                                    mckMessageService.processOnlineStatusChange(
                                         params.tabId,
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:6763]: does removal of isConversationInWaitingQueue doesn't effect anything?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 12]
```diff
@@ -5898,7 +6019,7 @@ var userOverride = {
                 err,
                 message
             ) {
-                if (_this.isFaqTabOpen() || _this.isConversationInWaitingQueue()) {
+                if (_this.isFaqTabOpen()) {
                     return;
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:6023]: here also?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 13]
```diff
@@ -10224,18 +10340,7 @@ var userOverride = {
...
-                if (contact && contact.metadata && contact.metadata.CONVERSATION_STATUS == Kommunicate.conversationHelper.status.WAITING ){
-                    // if conversation is in waiting queue
-                    displayName = MCK_LABELS['waiting.queue.message']['contact.name'];
-                    imgsrctag = '<div class="mck-videocall-image alpha_W"><span class="mck-contact-icon">...</span></div>';
-                    isConversationInWaitingQueue = true;
-                }
-                else{ 
-                    imgsrctag = _this.getContactImageLink(contact, displayName);
-                }
-
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:10238]: please see why these changes are getting removed from your code

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-widget-custom-events.js (hunk 7 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js:9]: These function definitions are getting overwritten by the user's custom event function definition, In case we call `sendEventToGoogleAnalytics(eventMapping['FAQ Menu'])` from here will be overwritten by the user code. And what if user also wants to send events to GoogleAnalytics as well as any other website?

Comment (devashishmamgain) [reply webplugin/js/app/km-widget-custom-events.js:9]: got it
found the right file, goto mck-sidebox-1.0.js

   _this.subscribeToEvents is a common function where all events are coming, i think here we should write code to call the GA related functions.
This is a common place for all events

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-widget-custom-events.js (hunk 7 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js:9]: So i think GA has zero dependencies on the chat widget custom events and `_this.subscribeToEvents` is called whenever the user call `Kommunicate.subscribeToEvents(events)` in on init function. I think this is not the right place to put GA-related code here?
What's your opinion?

Comment (devashishmamgain) [reply webplugin/js/app/km-widget-custom-events.js:9]: ya right

File: webplugin/js/app/kommunicate-ui.js (hunk 10 above)
Comment (Ojashwa) [reply webplugin/js/app/kommunicate-ui.js]: Rectified

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 12 above)
Comment (Ojashwa) [reply webplugin/js/app/mck-sidebox-1.0.js]: Rectified

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 13 above)
Comment (Ojashwa) [reply webplugin/js/app/mck-sidebox-1.0.js]: Rectified

File: webplugin/js/app/km-widget-custom-events.js (hunk 8 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js:5]: Rectified

File: webplugin/js/app/events/applozic-event-handler.js [hunk 14]
```diff
@@ -58,6 +58,11 @@ Kommunicate.KmEventHandler = {
         }
     },
     onMessageSent: function (message) {
+        window.kmWidgetEvents.sendEventToGoogleAnalytics(
```
Comment (devashishmamgain) [inline webplugin/js/app/events/applozic-event-handler.js:61]: is window. required?
see how else we can do it
some places we have kommunicateUI. and Kommunicate. as welll

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 15]
```diff
@@ -749,6 +749,10 @@ Kommunicate.richMsgEventHandler = {
         console.log('passenger detail submitted');
     },
     processQuickReplies: function (e) {
+        eventMapping.onRichMessageButtonClick.eventLabel = e.target.innerText;
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:752]: as onRichMessageButtonClick is global, so setting .eventLabel is problematic here.
what if at the same time some other event also happens then eventLabel value will be overriden

File: webplugin/js/app/km-widget-custom-events.js [hunk 16]
```diff
@@ -0,0 +1,179 @@
+/**
+ // Kommunicate custom events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+eventMapping = {
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:6]: add var, also better to put it under some object

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 17]
```diff
@@ -1889,6 +1892,71 @@ var userOverride = {
                     window.Applozic.ALSocket.events.onUserDeactivated =
                         events.onUserDeactivated;
                 }
+                if (typeof events.onChatWidgetOpen === 'function') {
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:1895]: instead of writing each and every event below, can something be done so that all events things become part of kmWidgetEvents in just few lines?

File: webplugin/js/app/km-widget-custom-events.js [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+        if (kommunicateCommons.isObject(eventObject)) {
+            applozic._globals.gaTrackingID &&
+                kmWidgetEvents.sendEventToGoogleAnalytics(
+                    eventObject.eventCateogry,
+                    eventObject.eventAction,
+                    eventObject.eventLabel,
+                    eventObject.eventValue
+                );
+            if (eventObject.eventLabel == 'FAQ Menu') {
+                window.kmWidgetEvents.onFaqClick(eventObject);
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:126]: what if we change
faqEvent: {
        eventCateogry: 'Kommunicate',
        eventAction: 'Click',
        eventLabel: 'FAQ Menu',
    }

to include the function name eg:

faqEvent: {
       data: {      
       eventCateogry: 'Kommunicate',
        eventAction: 'Click',
        eventLabel: 'FAQ Menu'
},
        eventFunc: window.kmWidgetEvents.onFaqClick
    }

and then here instead of multiple if condition for each event,
code we can write as:
eventObject.eventFunc(eventObject.data);

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
added few comments, review the whole code and figure out how to reduce the code

File: webplugin/js/app/km-widget-custom-events.js (hunk 18 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js]: In that case, if we need to call for every `eventMapping.faqEvent.eventFunc(eventObject.data);` in its respective function (where the base function is executing for the particular functionality) execution, can we do it like `kmWidgetEvents.onFaqClick(faqEvent.data)`?

Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js]: @devashishmamgain  pls refer to the above comment.

Comment (devashishmamgain) [reply webplugin/js/app/km-widget-custom-events.js]: it will be eventObject.eventFunc(eventObject.data)
and as eventFunc already have reference to onFaqClick and respective function, it will work by above line

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 19]
```diff
@@ -1892,71 +1892,32 @@ var userOverride = {
...
-                }
-                if (typeof events.onNotificationClick === 'function') {
-                    window.kmWidgetEvents.onNotificationClick =
-                        events.onNotificationClick;
-                }
-                if (typeof events.onRichMessageButtonClick === 'function') {
-                    window.kmWidgetEvents.onRichMessageButtonClick =
-                        events.onRichMessageButtonClick;
-                }
+                Object.keys(events) &&
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:1895]: is it possible to have something which sets it like:
window.kmWidgetEvents= events or something which extends only properties of events in kmWidgetEvents? is there any .extends something or something for that?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 20]
```diff
@@ -749,6 +749,8 @@ Kommunicate.richMsgEventHandler = {
         console.log('passenger detail submitted');
     },
     processQuickReplies: function (e) {
+        eventMapping.richMessageButtonEvent.data.eventLabel(e.target.innerText);
+        kmWidgetEvents.eventTracking(eventMapping.richMessageButtonEvent);
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:753]: can these 2 be combined to a single line, eg:
kmWidgetEvents.eventTracking(eventMapping.richMessageButtonEvent, e.target.innerText);
then have the eventTracking function handle it internally

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 19 above)
Comment (Ojashwa) [reply webplugin/js/app/mck-sidebox-1.0.js:1895]: can we use JQuery here?

Comment (devashishmamgain) [reply webplugin/js/app/mck-sidebox-1.0.js:1895]: no

File: webplugin/js/app/events/applozic-event-handler.js (hunk 14 above)
Comment (devashishmamgain) [inline webplugin/js/app/events/applozic-event-handler.js:61]: from here also, we can call kmWidgetEvents.eventTracking same as other places?

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/km-widget-custom-events.js [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+            eventAction: 'Open',
+            eventLabel: 'Chat Widget Open',
+        },
+        eventFunction: null,
+    },
+    onChatWidgetClose: {
+        data: {
+            eventCateogry: 'Kommunicate',
+            eventAction: 'Chat widget closed',
+            eventLable: 'Chat Widget Close',
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:63]: correct spelling eventLable
check the same at all places

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 22]
```diff
@@ -678,6 +681,12 @@ var userOverride = {
                       .removeClass('vis')
                       .addClass('n-vis')
                 : '';
+            document
+            .getElementById('chat-popup-widget-container')
+            .firstChild.removeEventListener(
+                'click',
+                KommunicateUI.captureGreetingMessageClick
```
Comment (shubhamtibra) [inline webplugin/js/app/mck-sidebox-1.0.js:688]: this event also seem to be twice

Comment (Ojashwa) [reply webplugin/js/app/mck-sidebox-1.0.js:688]: we are attaching click event when we found the greeting message `div `in DOM and removing it once the chat widget gets open.

File: webplugin/js/app/media/media-service.js [hunk 23]
```diff
@@ -6,6 +6,7 @@ Kommunicate.mediaService = {
         });
     },
     processMicClickedEvent: function () {
+        kmWidgetEvents.eventTracking(eventMapping.onVoiceIconClick);
```
Comment (shubhamtibra) [inline webplugin/js/app/media/media-service.js:9]: what will happen if there is error in user provided function? will it cause problems of further execution?

Comment (Ojashwa) [reply webplugin/js/app/media/media-service.js:9]: ratifed

File: webplugin/js/app/km-widget-custom-events.js [hunk 24]
```diff
@@ -0,0 +1,196 @@
...
+        // Any other analytics tool related code can be add here no need to paste it in every function
+        if (kommunicateCommons.isObject(eventObject)) {
+            if (customLabel) {
+                eventObject.data.eventLabel = customLabel;
+            }
+            if (customValue) {
+                eventObject.data.eventValue = customValue;
+            }
+            applozic._globals.gaTrackingID &&
+                kmWidgetEvents.sendEventToGoogleAnalytics(
```
Comment (shubhamtibra) [inline webplugin/js/app/km-widget-custom-events.js:35]: we are accessing this object in the definition itself. Does it work? Not sure about this, please verify once.

File: webplugin/js/app/km-widget-custom-events.js [hunk 25]
```diff
@@ -0,0 +1,196 @@
...
+                // checks if there is any errors in user provided function
+                try {
+                    eventObject.eventFunction(eventObject.data);
+                } catch (error) {
+                    console.log(error.stack);
+                }
+            }
+        } else {
+            throw new TypeError(
+                'eventTracking expect an object but got ' + typeof eventObject
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:51]: if this error is thrown, what will happen?
will it fail the chat widget functionality?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-widget-custom-events.js (hunk 24 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js:35]: pls check this fiddle- https://jsfiddle.net/csguj25o/

File: webplugin/js/app/km-widget-custom-events.js (hunk 25 above)
Comment (Ojashwa) [reply webplugin/js/app/km-widget-custom-events.js:51]: corrected

**Review by devashishmamgain — DISMISSED** [review]

**Review by shubhamtibra — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Review by devashishmamgain — DISMISSED** [review]

**Review by shubhamtibra — DISMISSED** [review]

File: webplugin/js/app/km-widget-custom-events.js [hunk 26]
```diff
@@ -0,0 +1,196 @@
...
+            eventCateogry: 'Kommunicate',
+            eventAction: 'Open',
+            eventLabel: 'Chat Widget Open',
+        },
+        eventFunction: null,
+    },
+    onChatWidgetClose: {
+        data: {
+            eventCateogry: 'Kommunicate',
+            eventAction: 'close',
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:81]: Close

File: webplugin/js/app/kommunicate-ui.js [hunk 27]
```diff
@@ -1139,11 +1147,30 @@ KommunicateUI = {
                 'km-animate',
                 'n-vis'
             );
-            if(kommunicateIframe.classList.contains("chat-popup-widget-vertical")){
-                var heightOfPopup = document.getElementById("chat-popup-widget-container").offsetHeight + 15;
-                var css = 'iframe#kommunicate-widget-iframe.chat-popup-widget-vertical { height:'+ heightOfPopup +'px!important;}';
-                parent.document.querySelector("style#kommunicate-style-sheet").innerText += css;
-            };
+            var greetingMessageContainer = document.getElementById(
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:1150]: looks like this code is duplicated in another file as well.

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-ui.js (hunk 27 above)
Comment (Ojashwa) [reply webplugin/js/app/kommunicate-ui.js:1150]: we are attaching an event listener to the greeting message `div` and removing it in another file where the chat widget gets open.

Comment (devashishmamgain) [reply webplugin/js/app/kommunicate-ui.js:1150]: greetingMessageContainer.firstChild.removeEventListener(
                    'click',
                    KommunicateUI.captureGreetingMessageClick
                );

**Review by devashishmamgain — DISMISSED** [review]

**Review by shubhamtibra — DISMISSED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by shubhamtibra — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]
